eeps/eep-0066.md

```diff
@@ -0,0 +1,432 @@
+  according to the regular expression rules, and in addition to them
+  the [end delimiter][] character can be escaped with a «`\`» character.
+  Between triple-quote delimiters according to [EEP 64][]
+  there is no end delimiter character to escape.
+
+### String Delimiters
+
+Immediately following the [Sigil Type][] is the string start delimiter.
+A specific start delimiter character has a corresponding
+end delimiter character.
+
+These are the start delimiter characters and they have themselves
+as end delimiter: single quote «`'`», and double quote «`"`».
+
+Triple-quote delimiters are also allowed, that is; a sequence of
+3 or more double quote «`"`» characters as described in [EEP 64][].
+
+### String Content
+
+Between the start and end [String Delimiters][], all characters
+are string content.
+
+In a triple-quoted string all characters are verbatim, but stripping
+of indentation and leading and trailing newline is done as usual
+as described in [EEP 64][].
+
+In a string with single character [String Delimiters][],
+normal Erlang escape sequences prefixed with «`\`» are honoured,
+as usual for regular Erlang strings and quoted atoms
+
+A specific [Sigil Type][] can have it's own character escaping rules,
+which may affect finding the [end delimiter][].
+
+### Sigil Suffix
+
+Immediately following the String Content comes the Sigil Suffix,
+that may be empty.
+
+The Sigil Suffix consists, as the [Sigil Type][] in the [Sigil][] prefix,
+of name characters.
+
+The Sigil Suffix may indicate how to interpret the String Content,
+for a specific [Sigil Type][].  For example; for the «`~r`» [Sigil][]
+(regular expression), the Sigil Suffix is interpreted as short form
+compile options such as «`i`» that makes the regular expression
+character case insensitive.
+
+Things that may have to be performed by the tokenizer, such as
+how to handle escape character rules, should not be affected
+by the Sigil Suffix, since the tokenizer has already scanned
+the [String Content][] when it sees the Sigil Suffix.
+
+### Regular Expressions
+
+#### Uncompiled Regular Expression
+
+The value of a regular expression [Sigil][] is a 2-tuple
+with the uncompiled regular expression and its compile flags
+(in the guise of a sequence of character flags).
+
+With this representation, the `re` module can be augmented
+with functions that accept this tuple format.  These functions
+are `re:compile/1,2`, `re:replace/3,4` `re:run/2,3`,
+and `re:split/2,3`  Translation of the compile flag characters
+into `re:compile_option()`s should be done by these functions.
+
+Example:
+
+    1> re:run("ABC123", ~r"abc\d+"i, [{capture,first,list}]).
+    {match,["ABC123"]}
+
+Since the [Sigil][] value represents an uncompiled regular expression,
+the user can choose when to compile it with `re:compile/1,2`,
+or use it directly in e.g `re:run/2,3`.
+
+It is possible to implement an optimization to make the compiler
```

Comment:
This is a nice addition but not a must for the EEP.
But good that it is documented.